packages/peregrine/src/Router/Router.js

```diff
@@ -1,33 +1,49 @@
-import React, { Component, createContext } from 'react';
+import React, { Component } from 'react';
 import { BrowserRouter, Route } from 'react-router-dom';
 import { func, object, string } from 'prop-types';
-
-export const { Consumer, Provider } = createContext();
+import MagentoRouteHandler from './MagentoRouteHandler';
 
 export default class MagentoRouter extends Component {
     static propTypes = {
         apiBase: string.isRequired,
         routerProps: object,
-        using: func // e.g., BrowserRouter, MemoryRouter
+        using: func, // e.g., BrowserRouter, MemoryRouter
+        renderRoutes: func,
+        renderMagentoRoutingError: func
     };
 
     static defaultProps = {
         routerProps: {},
-        using: BrowserRouter
+        using: BrowserRouter,
+        renderRoutes: ({ magentoRoute }) => magentoRoute
     };
 
     render() {
-        const { apiBase, children, routerProps, using: Router } = this.props;
+        const {
+            apiBase,
+            routerProps,
+            using: Router,
+            renderRoutes,
+            renderMagentoRoutingError
+        } = this.props;
 
         return (
             <Router {...routerProps}>
-                <Route>
-                    {routeProps => (
-                        <Provider value={{ apiBase, ...routeProps }}>
-                            {children}
-                        </Provider>
```

Comment:
The purpose of the recent routing changes was to provide the route info as context, which avoids render blocking issues and allows us to render route-based views at any level of the app. This would completely undo that work.

Reply:
Official react router documentation provides a solution for the issue of blocked updates https://reacttraining.com/react-router/core/guides/redux-integration/blocked-updates and I added it. In my opinion, context api increased complexity of the code and it is possible to implement the same thing easier.

Reply:
@jimbo What do you think?

Reply:
I disagree. Context doesn't increase the complexity of the code, and it certainly doesn't need to be removed here.